lib/InputWithButton/index.js

```diff
@@ -1,8 +1,32 @@
 import React, { Component } from 'react';
-import cx from 'classnames';
+import styled from 'styled-components';
 import PropTypes from 'prop-types';
+import Input from '../core/Input';
 import Button from '../Button';
 
+const Container = styled.div`
+  display: flex;
+  flex-direction: row;
+`;
+
+const LeftHandInput = styled(Input)`
+  width: 220px;
+  border-radius: 4px 0px 0px 4px;
+`;
+
+const RightHandButton = styled(Button)`
+  ${props =>
+    props.paddingSmall
```

Comment:
How do we handle multiple modifiers?

```
styles(Button)`
  ${props =>
    props.paddingSmall
      ? '' : '',
    props.paddingLarge
      ? '' : ''
   }
`

Does the function take multiple arguments?
```

Reply:
Whatever is returned from the function is added as a value in the string literal (just as in normal JS), eg.

```
styled(Button)`
  padding: ${props => props.paddingSmall ? '5' : '10'}px; 
`
```
is essentially just 
```
styled(Button)`
  padding: ${props => {
      return props.paddingSmall ? '5' : '10'
   }px; 
`
```
and you can call the function as many times as you like in the string literal
```
styled(Button)`
  padding: ${props => props.paddingSmall ? '5' : '10'}px; 
  width: ${props => props.fullWidth ? '100%' : '500px'}; 
`
```
I don't think this is what you are asking though, so ping me back!